### PR TITLE
Team Teaching - Move overlay rendering to Scene pipeline

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

File: interface/src/ui/overlays/Overlays.cpp [hunk 1]
```diff
@@ -31,6 +32,50 @@
 #include "TextOverlay.h"
 #include "Text3DOverlay.h"
 
+
+namespace render {
+    template <> const ItemKey payloadGetKey(const Overlay::Pointer& overlay) {
+        if (overlay->is3D() && !static_cast<Base3DOverlay*>(overlay.get())->getDrawOnHUD()) {
+            if (static_cast<Base3DOverlay*>(overlay.get())->getDrawInFront()) {
+                return ItemKey::Builder().withTypeShape().withNoDepthSort().build();
```
Comment (huffman) [inline interface/src/ui/overlays/Overlays.cpp]: NO_DEPTH_SORT is not yet implemented - Sam will be working on that, until then some overlays will incorrectly render behind other objects.

**Comment by huffman** [comment]
Hold off on this for one moment

**Comment by hifi-gustavo** [comment]
Build console output for commit: b4db787837161b5a45c249f7c164973befb38e07 available at: https://gist.github.com/2fbcc094188a21c3db95<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5000.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5000.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 667dccbb874ee299f7c3e8f136c688db33fb49fc available at: https://gist.github.com/2fbcc094188a21c3db95<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5000.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5000.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b37a02304191e55c4860e9d2d82c238f65241b5f available at: https://gist.github.com/2fbcc094188a21c3db95<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5000.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5000.dmg

**Pull request merged**